### PR TITLE
(#1691) adds usa section page to docs site

### LINKED ISSUE #1691: Create documentation for usa-section
## As a documentation site user, I want to see documentation on `usa-section`, so that I can incorporate it correctly on my website.

Create page for usa-section within the Components section of the documentation site with the content needed for NCIDS launch.

### Tasks
- [x] Create content in shared Word Document `<insert link to shared folder>`
    - [x] IA responsibility:
    - [x] Dev responsibility:
    - [x] Design responsibility:
- [x] Review content in shared document
- [x] Create images in shared folder: See comment below.
- [ ] Convert content and images to markdown
- [ ] Add to documentation site

### PR DESCRIPTION
**Pull request by dlescarbeau**
Closes #1691

## Pull Request Details

_Add description_

Closes #<TICKET-NUMBER>

## Author PR Checklist
Items that the author of the PR is responsible for checking before submitted the PR.

### General:
- [x] I have reviewed the acceptance criteria defined in the ticket and ensured the work has been completed.
- [x] The commit message passes [all quality commit message standards](https://github.com/NCIOCPL/ncids/wiki/Creating-Quality-Commit-Messages).
- [x] Unit tests have been updated or created to reflect any javascript changes.
- [x] Storybook scenarios have been updated or created to reflect any html/css/js changes.

### Accessibility:
- [x] WCAG 2.1 Level AA requirements have been met.

### Development:
- [x] Any new or updated javascript code has 100% unit test coverage.
- [x] New or updated breakpoints have regression images.
- [x] Breaking changes have been thoroughly documented in the PR.

## Product Reviewer PR Checklist
Items the product team is responsible for reviewing.

### General:
- [x] There are no unexpected or unapproved regression image changes. 
- [x] Functionality of interactive elements meet the acceptance criteria.
- [x] The product is visually and functionally the same across the different browsers.

### Accessibility:
- [x] AxeDev Tools: there are no new or outstanding accessibility issues introduced in this PR.
- [x] Lighthouse: scores have not noticeably decreased during this PR.
- [x] Wave: there are no new errors or contrast errors introduced in this PR.

## Design Reviewer PR Checklist
Items the design team is responsible for reviewing.  

### General:
- [x] New or updated features introduced in this PR are developed mobile-first.
- [x] Breakpoint changes and regression images match those breakpoints.
- [x] This PR has been tested in all supported browsers at all breakpoints.

## Developer Reviewer PR Checklist
Items the development team is responsible for reviewing.

### General:
- [x] New code passes code quality standards set by industry standards.
- [x] The expected Storybook stories have been added or updated for the new or updated feature.
- [ ] The expected unit tests have been added or updated for the new or updated feature.

### Accessibility:
- [x] VoiceOver: Described content matches with what was expected.
- [x] Keyboard navigation: new or updated features and content are navigable via the keyboard.

### CODE REVIEW

**Review by sarinapadilla — CHANGES_REQUESTED** [review]
Please make these updates:
* The page title, browser title, and nav label should be "Section" rather than "USA Section"
* The "Variations" heading should be "Modifications"

**Review by karlikpj — APPROVED** [review]

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Three main things hit me:
1. The `<section>` elements should not be `<section>` elements without a `aria-label` or `aria-labelledby`. This should be noted in the accessibility. If there is no label for the component, then a `<div>` element should be used instead.
2. Are we not showing off the look of the two variants.
3. We really should talk about the behavior when 2 same-themed sections are next to each other.

**Comment by andyvanavery31** [comment]
In addition to Bryan's points:
- We need a `<ul>` at the parent level of the unordered list in the [Modifications section](https://designsystem-dev.cancer.gov/pr-1862/components/section#modifications) in the table.
- The table heading under [Modifications](https://designsystem-dev.cancer.gov/pr-1862/components/section#modifications) should be "Modifier" not "Variant" to match the section and to match the rest of the component pages

**Review by sarinapadilla — APPROVED** [review]
Looks good to product!

File: docs/content/components/section.mdx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    - title: Default Section
+      intro: |
+        * The section component will default to be the full width of the component's container, without margins or horizontal padding.
+        * A component that covers the full width of the container without margins would likely be a direct child element of the section component.
+      preview: true
+      code: |
+        <div class="usa-section">
+          This is example text in a default section. Notice the lack of margins and the padding on the top and bottom of the section. Your HTML would go here in place of this text.
+        </div>
+    - title: Applying Standard Page Margins
```
Comment (bryanpizzillo) [inline docs/content/components/section.mdx:74]: In testing with other developers, the `<section>` examples were missing. Please add 2 examples above the Applying Standard Page Margins example for:
* Making an Accessible Section using a Label
* Making an Accessible Section Labelled with Another Element

Feel free to use your discretion in adding good descriptions for each.

File: docs/content/components/section.mdx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      intro: |
+        * A grid-container class should be applied to child elements of the section component that need standard margins applied.
+        * By using this class, auto margins will be applied, and a max-width will be given to the element based on our standard grid-container width. This will also apply a small amount of padding-left/right to the element.
+        * Components that would normally fall into the standard content container would utilize this structure.
+      preview: true
+      code: |
+        <section class="usa-section" aria-label="a section with a grid container">
```
Comment (bryanpizzillo) [inline docs/content/components/section.mdx:81]: Change this back to a div now that we will add other examples.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]

**Review by olitharp-nci — APPROVED** [review]

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**